### PR TITLE
DBZ-5859 first draft of reactive outbox docs

### PR DESCRIPTION
**Pull request by ingmarfjolla**
https://issues.redhat.com/browse/DBZ-5859

### CODE REVIEW

**Comment by jpechane** [comment]
@Naros Could you please take a look at the R and when fine from your PoV pass it to @roldanbob? Thanks

File: documentation/modules/ROOT/pages/integrations/outbox.adoc [hunk 1]
```diff
@@ -124,6 +124,55 @@ The {prodname} CDC connector in conjunction with the Outbox Event Router will be
 
 To see a full end-to-end demo, the https://github.com/debezium/debezium-examples/tree/main/outbox[Outbox] example illustrates two Quarkus microservice applications using the outbox pattern to share data between them when orders are placed or cancelled.
 
+[id=reactive-outbox]
+== Reactive Variant
+
+If your application is making use of Reactive Datasources or Hibernate Reactive, then there are some differences.
```
Comment (roldanbob) [inline documentation/modules/ROOT/pages/integrations/outbox.adoc:130]:
```suggestion
If your application uses reactive datasources, or Hibernate Reactive, you must use a slightly different configuration to add the extension to your application. 
```

File: documentation/modules/ROOT/pages/integrations/outbox.adoc [hunk 2]
```diff
@@ -124,6 +124,55 @@ The {prodname} CDC connector in conjunction with the Outbox Event Router will be
 
 To see a full end-to-end demo, the https://github.com/debezium/debezium-examples/tree/main/outbox[Outbox] example illustrates two Quarkus microservice applications using the outbox pattern to share data between them when orders are placed or cancelled.
 
+[id=reactive-outbox]
+== Reactive Variant
+
+If your application is making use of Reactive Datasources or Hibernate Reactive, then there are some differences. 
+
+To import the reactive variant, the extension is added as follows:
```
Comment (roldanbob) [inline documentation/modules/ROOT/pages/integrations/outbox.adoc:132]:
```suggestion
For example, use the following configuration to  import the reactive variant of the extension:
```

File: documentation/modules/ROOT/pages/integrations/outbox.adoc [hunk 3]
```diff
@@ -124,6 +124,55 @@ The {prodname} CDC connector in conjunction with the Outbox Event Router will be
...
+[source,xml,subs="verbatim,attributes"]
+----
+<dependency>
+  <groupId>io.debezium</groupId>
+  <artifactId>debezium-quarkus-outbox-reactive</artifactId>
+  <version>{debezium-version}</version>
+</dependency>
+----
+
+The extension provides the application with the `io.debezium.outbox.quarkus.ExportedEvent` interface just like in the non reactive variant, but also comes with the `DebeziumOutboxHandler` class which can be injected into your applications and exposes the `persistToOutbox` method. One example invocation is seen below which is using the same `ExportedEvent` as the previous example:
```
Comment (roldanbob) [inline documentation/modules/ROOT/pages/integrations/outbox.adoc:143]:
```suggestion
The reactive extension provides the application with the same `io.debezium.outbox.quarkus.ExportedEvent` interface as the non-reactive variant, but it also provides the `DebeziumOutboxHandler` class.
This class can be injected into your applications and expose the `persistToOutbox` method. 
The following example shows an invocation that uses the same `ExportedEvent` as in the earlier example:
```

File: documentation/modules/ROOT/pages/integrations/outbox.adoc [hunk 4]
```diff
@@ -124,6 +124,55 @@ The {prodname} CDC connector in conjunction with the Outbox Event Router will be
...
+        return orderRepository.persistAndFlush(order)
+        .call(debeziumOutboxHandler.persistToOutbox(new OrderCreatedEvent(Instant.now(), order)));
+
+    }
+}
+----
+
+[NOTE]
+====
+The `persistToOutbox` method returns a Uni, so however you decide to invoke it you must subscribe to the Uni in order to invoke it. More material on Reactive and Mutiny can be found link:https://quarkus.io/guides/mutiny-primer[here]
```
Comment (roldanbob) [inline documentation/modules/ROOT/pages/integrations/outbox.adoc:171]:
```suggestion
The `persistToOutbox` method returns a Uni, so however you decide to invoke it you must subscribe to the Uni in order to invoke it. 
For more information about using the Mutiny library to build reactive applications, see the link:https://quarkus.io/guides/mutiny-primer[Quarkus Mutiny primer].
```

Comment (roldanbob) [reply documentation/modules/ROOT/pages/integrations/outbox.adoc:171]: The part about invoking the `persistToOutbox` method seems a little circular ("invoke it,  ... subscribe to the [returned] Uni ... to invoke it"), but I don't know enough about the subject to make the sentence more clear.

**Review by roldanbob — CHANGES_REQUESTED** [review]
Made a few suggestions. Per my comment, I wasn't sure how to interpret the note about the `persistToOutbox` method.

File: documentation/modules/ROOT/pages/integrations/outbox.adoc (hunk 3 above)
Comment (Naros) [reply documentation/modules/ROOT/pages/integrations/outbox.adoc:143]: In effect, @roldanbob, there is a lot of "wiring" required when you're working with reactive modules, and so inside Quarkus you can inject the `DebeziumOutboxHandler` bean into your application, and it provides you directly with a method that allows you to quickly persist the `ExportedEvent` to the outbox table without needing to be concerned about how to wire all the dependencies required to do this.

File: documentation/modules/ROOT/pages/integrations/outbox.adoc (hunk 4 above)
Comment (roldanbob) [reply documentation/modules/ROOT/pages/integrations/outbox.adoc]: Thanks, @Naros. Your latest change addresses my concerns.

File: documentation/modules/ROOT/pages/integrations/outbox.adoc (hunk 3 above)
Comment (roldanbob) [reply documentation/modules/ROOT/pages/integrations/outbox.adoc]: I made a slight change to my original edit that incorporates information from your comment to improve the clarity of the paragraph.

**Review by Naros — APPROVED** [review]

**Comment by Naros** [comment]
Applied to main (=2.2).

**Pull request merged**